### PR TITLE
Discard signature_check job,check image signature directly

### PR DESCRIPTION
**Pull request by wewang58**
[wewang@fedora release-tests]$ python cli -r 4.13.2 image-signed-check 
2023-06-21T09:01:18Z: INFO: task [Payload image is well signed] status is changed to [In Progress]
sha256:6ef3cf4bed1970d547dce08a6e334b675d361b212xxxxxxxxxxxxxxxxxxx
Comparing digest from url: https://mirror.xxxxxx.com/pub/xxxxx-xx/signatures/xxx/release/sha256=6ef3cf4bed1970d547dcxxx4b675d361b2xxxxxxxxxxxxxxxxxxxxxxxxx/
request status code 200
Signature check PASSED
2023-06-21T09:01:25Z: INFO: task [Payload image is well signed] status is changed to [Pass]

### CODE REVIEW

**Comment by wewang58** [comment]
@rioliu-rh  Please review it, thanks

File: oar/cli/cmd_image_signed_check.py [hunk 1]
```diff
@@ -1,41 +1,45 @@
...
-            logger.info("job[signature_check] already triggered and in progress, not need to trigger again")
+    report = WorksheetManager(cs).get_test_report()
+    image_signed_check_result = report.get_task_status(LABEL_TASK_PAYLOAD_IMAGE_VERIFY)
+    if image_signed_check_result == TASK_STATUS_PASS:
+        logger.info("image signed check already pass, not need to trigger again")
+    else:
+        report.update_task_status(LABEL_TASK_PAYLOAD_IMAGE_VERIFY, TASK_STATUS_INPROGRESS)
+        res = requests.get(cs.get_release_url() + "/releasestream/4-stable/release/" + cs.release)
+        if res.status_code != 200:
+            print(res.status_code, res.reason)
```
Comment (rioliu-rh) [inline oar/cli/cmd_image_signed_check.py:27]: @wewang58 don't suggest to use print to output. logger is recommended

File: oar/cli/cmd_image_signed_check.py [hunk 2]
```diff
@@ -1,41 +1,45 @@
...
+        # 1. get the digest
+        digest_sha = soup.find(string=re.compile("sha256:"))
+        if len(digest_sha) == 0:
+            logger.error("No image digest found!")
+        print(digest_sha)
+        reformatted_digest = digest_sha.replace(':', '=')
+        # 2. query the mirror location
+        mirror_url = cs.get_signature_url() + reformatted_digest + "/"
+        print(f'Comparing digest from url: {mirror_url}')
+        res = requests.get(mirror_url)
```
Comment (rioliu-rh) [inline oar/cli/cmd_image_signed_check.py:38]: @wewang58 need to catch exception raised from 3rd party library, then mark the task as failed

Comment (wewang58) [reply oar/cli/cmd_image_signed_check.py]: @rioliu-rh  Updated, please check again, thanks

File: oar/cli/cmd_image_signed_check.py [hunk 3]
```diff
@@ -1,41 +1,53 @@
...
+        report.update_task_status(LABEL_TASK_PAYLOAD_IMAGE_VERIFY, TASK_STATUS_INPROGRESS)
+        try:
+            res = requests.get(cs.get_release_url() + "/releasestream/4-stable/release/" + cs.release)
+            res.raise_for_status()
+            soup = BeautifulSoup(res.content, 'lxml')
+            # 1. get the digest
+            digest_sha = soup.find(string=re.compile("sha256:"))
+            if len(digest_sha) == 0:
+                logger.error("No image digest found!")
+            logger.info(digest_sha)
```
Comment (rioliu-rh) [inline oar/cli/cmd_image_signed_check.py:34]: suggest to refine this message: like `f"found image digest: ${digest_sha}"`

File: oar/cli/cmd_image_signed_check.py [hunk 4]
```diff
@@ -1,41 +1,53 @@
...
+            # 2. query the mirror location
+            mirror_url = cs.get_signature_url() + reformatted_digest + "/"
+            try:
+                logger.info(f'Comparing digest from url: {mirror_url}')
+                rest = requests.get(mirror_url)
+                rest.raise_for_status()
+                if rest.status_code == 200:
+                    logger.info("Signature check PASSED")
+                    report.update_task_status(LABEL_TASK_PAYLOAD_IMAGE_VERIFY, TASK_STATUS_PASS)
+            except RequestException:
```
Comment (rioliu-rh) [inline oar/cli/cmd_image_signed_check.py:45]: we don't need to catch exception here

Comment (wewang58) [reply oar/cli/cmd_image_signed_check.py]: @rioliu-rh Updated new comments and tested, please check again, thanks

**Pull request merged**